rust/common-tracing/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+#![warn(missing_docs)]
+
+//! A library containing shared tracing functionality across common crates.
+
+#[cfg(target_arch = "wasm32")]
+mod inner {
+    use std::sync::Once;
+    static INITIALIZE_TRACING: Once = Once::new();
+
+    /// Initialize tracing-based logging.
+    pub fn initialize_tracing() {
+        INITIALIZE_TRACING.call_once(|| {
+            console_error_panic_hook::set_once();
+            tracing_wasm::set_as_global_default();
+        })
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+mod inner {
+    use anyhow::Result;
+    use std::sync::Once;
+    use tracing_subscriber::{EnvFilter, FmtSubscriber};
+
+    static INITIALIZE_TRACING: Once = Once::new();
+
+    /// Initialize tracing-based logging.
+    pub fn initialize_tracing() {
```

Comment:
I really, really want to avoid going down this path again where we have a totalizing entrypoint for initializing tracing. The value of a shortcut like this is for turning on logging in tests, and we probably never want to configure tracing this way in code that may end up in production.

So, with the above in mind, can you factor the critical tracing layer configuration into its own helper and get us started on the right track by manually setting up a subscriber in `builder.rs`?

Reply:
SG, a common `Layer` could both be used in tests via e.g. `#[common_trace]` and set up a subscriber, and manually in binaries with whatever subscriber